src/main/java/HttpProtocal.java

```diff
@@ -0,0 +1,8 @@
+package com.td.HttpServer;
+
+public interface HttpProtocal {
```

Comment:
So, whether you change the way requests are created or you leave it as-is, it isn't too hard to create a request however you want for testing purposes. With your current implementation, for example:

```java
Request badRequest = new Request("NONSENSE /foo HTTP/1.1");
Request goodRequest = new Request("GET / HTTP/1.1");
```

Interfaces are great when you want to swap in different kinds of the same general thing, but in this domain there's only a single thing—an HTTP request. Whether a given request is valid or invalid is a property of that instance of the request, not a different _type_